### DISCUSSION TITLE
Per story controls/argTypes?

### DISCUSSION THREAD

**Discussion by fredboyle** [storybookjs/storybook#17752] [Help]
Is there a mechanism to add custom controls (argTypes) on a per story basis?

Some stories have a specific wrapper or other Storybook specific mechanism I'd like to control with the delightful controls baked in. However I don't need it for every story in a file, only a small select few.

I can add them to the `default` of the story and control using ```foo.parameters = {
  controls: { exclude: ['arg1'] },
};``` however that doesn't work for the primary story since any args/controls visible on the primary story will also render in the argsTable in the "Docs" view.

Any help greatly appreciated.

**Comment by LucidityDesign** [comment]
With [CSF3](https://storybook.js.org/blog/component-story-format-3-0/) you can do something like this:
```
export default {
  component: Button,
  argTypes: {
    disabled: {
      control: "boolean",
    },
  },
} satisfies ComponentMeta<typeof Button>
...
export const AnotherStory = {
  argTypes: {
    disabled: {
      control: false,
    },
  },
  render: () => (<>),
} satisfies Story
```
